### ISSUE TITLE
Passwords are not usable

### ISSUE THREAD

**Issue by Mrs-Feathers** [heviat/Mailu#4]
### Environment

- [x] docker compose
- [ ] kubernetes
- [ ] docker swarm

### Version

- Version: `master`

## Description

Passwords aren't able to be used on the webui at all. only OIDC works, but passwords are needed for client logins. everything else works perfect, and logging in with an external client works too if you use the docker command to set a password, but you can't change it via the webui or anything.. it throws an error.

## Replication Steps

1. create a new account via OIDC login. 
2. now you have an account with no password and no way to set it.. so you need admin help to login with a client.

## Observed behaviour

passwords are still able to be set with the docker command, so with an admin's help users are still able to set a random password. but doing this is considered a very high security flaw. even when the account's password is set, changing it via the webui from the user's account results in a default 500 error.

## Expected behaviour

A user should be able to set and edit their own password. or at least a user should be able to set their previously unset password so they can use a client. or even if an admin needs to create a temporary password, they should be able to change it.

## Logs

```
mailu-admin-1      | [2023-10-24 19:41:51,972] ERROR in app: Exception on /admin/user/password [POST]
mailu-admin-1      | Traceback (most recent call last):
mailu-admin-1      |   File "/app/venv/lib/python3.11/site-packages/flask/app.py", line 2529, in wsgi_app
mailu-admin-1      |     response = self.full_dispatch_request()
mailu-admin-1      |                ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
mailu-admin-1      |   File "/app/venv/lib/python3.11/site-packages/flask/app.py", line 1825, in full_dispatch_request
mailu-admin-1      |     rv = self.handle_user_exception(e)
mailu-admin-1      |          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
mailu-admin-1      |   File "/app/venv/lib/python3.11/site-packages/flask_restx/api.py", line 674, in error_router
mailu-admin-1      |     return original_handler(e)
mailu-admin-1      |            ^^^^^^^^^^^^^^^^^^^
mailu-admin-1      |   File "/app/venv/lib/python3.11/site-packages/flask/app.py", line 1823, in full_dispatch_request
mailu-admin-1      |     rv = self.dispatch_request()
mailu-admin-1      |          ^^^^^^^^^^^^^^^^^^^^^^^
mailu-admin-1      |   File "/app/venv/lib/python3.11/site-packages/flask/app.py", line 1799, in dispatch_request
mailu-admin-1      |     return self.ensure_sync(self.view_functions[rule.endpoint])(**view_args)
mailu-admin-1      |            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
mailu-admin-1      |   File "/app/venv/lib/python3.11/site-packages/flask_login/utils.py", line 290, in decorated_view
mailu-admin-1      |     return current_app.ensure_sync(func)(*args, **kwargs)
mailu-admin-1      |            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
mailu-admin-1      |   File "/app/mailu/ui/access.py", line 29, in wrapper
mailu-admin-1      |     return callback(function, args, kwargs, dargs, dkwargs)
mailu-admin-1      |            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
mailu-admin-1      |   File "/app/mailu/ui/access.py", line 20, in callback
mailu-admin-1      |     return function(*args, **kwargs)
mailu-admin-1      |            ^^^^^^^^^^^^^^^^^^^^^^^^^
mailu-admin-1      |   File "/app/mailu/ui/views/users.py", line 130, in user_password_change
mailu-admin-1      |     return _process_password_change(forms.UserPasswordChangeForm(), user_email)
mailu-admin-1      |            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
mailu-admin-1      |   File "/app/mailu/ui/views/users.py", line 112, in _process_password_change
mailu-admin-1      |     elif user_email or models.User.login(user_email_or_current, form.current_pw.data):
mailu-admin-1      |                        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
mailu-admin-1      |   File "/app/mailu/models.py", line 724, in login
mailu-admin-1      |     return user if (user and user.enabled and user.check_password(password)) else None
mailu-admin-1      |                                               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
mailu-admin-1      |   File "/app/mailu/models.py", line 616, in check_password
mailu-admin-1      |     return self.is_authenticated()
mailu-admin-1      |            ^^^^^^^^^^^^^^^^^^^^^^^
mailu-admin-1      | TypeError: 'bool' object is not callable
```

**Comment by Mrs-Feathers** [comment]
i report this here because i am not able to see this issue in normal mailu. i believe this is a bug specific for Mailu-OpenID.

Also, I thank you so much for creating this as every other aspect and option works absolutely perfectly. so much easier and better than LDAP. I use Authentik to power SSO for my self-hosted server and LDAP doesn't work that great on there so this option is a life saver!

**Comment by Mrs-Feathers** [comment]
I have solved this issue however i wish for the author to see this solution as i feel it could benefit everyone and make this more usable. 

Solution:

1. I commented out the "update password" link in /app/mailu/ui/templates/sidebar.html so users can't see it and thus not be confused because its not supposed to work with OIDC.

2. I set the "*******" string in /app/mailu/ui/templates/client.html to something useful telling the user they should use a token. the text itself is a simple a href link to the authorization tokens page.

solved.  i guess i could comment out the login form on the front page too, as passwords don't work so theres no way you can log in anyway and thats the one place tokens don't work either. so i recommend to also comment this out.

![Bildschirmfoto 2023-10-24 um 20 50 50](https://github.com/heviat/Mailu-OpenID/assets/1746550/ac0a0d42-c4f7-4370-8227-054c6a80151d)

**Comment by Encotric** [comment]
Hi, thank you for your help! Yeah, authorization tokens are the way to go for mail clients, because the Mailu server does not know the actual password of the user (as it is stored and validated by the OpenID Connect provider). But I got your point about the password change form and client setup page, so I released a few optimizations. It is now possible to set a custom password change URL for OIDC-registered users or completely remove the password change entry in the sidebar, as you did. 
Additionally, the client setup page shows a similar notice to your modification when OIDC-registered users are logged in.

I hope this helps!
